backend-tests/tests/test_decommission.py

```diff
@@ -0,0 +1,275 @@
+
+        # check device is rejected
+        r = dc.with_auth(dtoken).call('GET',
+                                      deployments.URL_NEXT,
+                                      qs_params={'device_type': 'foo',
+                                                 'artifact_name': 'bar'})
+        assert r.status_code == 401
+
+        # check that the workflow completed successfully
+        timeout = time.time() + (60 * 5)
+        while time.time() < timeout:
+            wfs = conductor.get_workflows(params=query)
+            logger.debug("WORKFLOWS: ", wfs)
+            if wfs['totalHits'] == initial_wfs['totalHits'] + 1:
+                workflow = wfs['results'][0]
+                m = re.search('(?<=device_id=)\w+', workflow['input'])
+                devId = m.group(0)
+                if workflow['status'] == "COMPLETED" and devId == aset.did:
+                    logger.debug("decommissioning workflow completed")
+                    break
+                else:
+                    logger.debug("waiting for decommission_device workflow...")
+                    time.sleep(1)
+            else:
+                logger.debug("waiting for decommission_device workflow...")
+                time.sleep(1)
+        else:
+            assert False, "decommission_device workflow didn't complete for [%s]" % (aset.did,)
+
+        # check device gone from inventory
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            r = invm.with_auth(utoken).call('GET',
+                                            inventory.URL_DEVICE,
+                                            path_params={'id': aset.did})
+            if r.status_code == 404:
+                break
+            else:
+                logger.debug("waiting for the device to be removed from inventory...")
+                time.sleep(1)
+        else:
+            assert False, "device not removed from the inventory"
+
+        # check device gone from deviceauth
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            r = devauthm.with_auth(utoken).call('GET',
+                              deviceauth_v2.URL_DEVICE.format(id=aset.did))
+            if r.status_code == 404:
+                break
+            else:
+                logger.debug("waiting for the device to be removed from deviceauth...")
+                time.sleep(1)
+        else:
+            assert False, "device not removed from the deviceauth"
+
+#class TestDeviceDecomissioning(TestDeviceDecomissioningBase):
+#    def test_ok(self, user, devices):
+#        self.do_test_ok(user, devices[0])
+
+class TestDeviceDecomissioningEnterprise(TestDeviceDecomissioningBase):
+    def test_ok(self, tenants_users_devices):
+        t = tenants_users_devices[0]
+        self.do_test_ok(user=t.users[0], device=t.devices[0], tenant_token=t.tenant_token)
+
+        t1 = tenants_users_devices[1]
+        self.verify_devices_unmodified(t1.users[0], t1.devices)
+
```

Comment:
As it turned to result in a single test case, maybe we can just remove the classes and just have two helpers (do_test_ok, verify_devices_unmodified) + one test case calling them.

Reply:
I've just followed the convention we have in the backend integration tests.
Should we change it everywhere in the tests?

Reply:
It's okey. Let's follow the current convention :+1: